graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -1035,31 +1116,39 @@ def enter_field_definition(
         path: List[Any],
         ancestors: List[Any],
     ) -> None:
-        """Check that no type Bar contains a field of type Foo, where Foo is suppressed."""
-        if self.current_type == self.query_type:
-            return IDLE
-        # At a field of a type that is not the query type
-        field_name = node.name.value
-        field_type = get_ast_with_non_null_and_list_stripped(node.type).name.value
-        if self.type_renamings.get(field_type, field_type):
-            return IDLE
-        # Reaching this point means this field is of a type to be suppressed.
+        """Check that no type Bar contains a field of type Foo, where Foo is unqueryable."""
         if self.current_type is None:
-            raise AssertionError(
-                "Entered a field not in any ObjectTypeDefinition scope because "
-                "self.current_type is None"
-            )
-        if self.current_type == field_type:
-            # Then node corresponds to a field belonging to type T that is also of type T.
-            # Therefore, we don't need to explicitly suppress the field as well and this should not
-            # raise errors.
+            # Do nothing if at fields in interfaces because interfaces cannot be suppressed for now.
             return IDLE
-        if self.field_renamings.get(self.current_type, {}).get(field_name, {field_name}) == set():
-            # Field was also suppressed so this should not raise errors.
+        current_type_name = self.current_type.name.value
+        if current_type_name == self.query_type:
             return IDLE
-        if self.current_type not in self.fields_to_suppress:
-            self.fields_to_suppress[self.current_type] = {}
-        self.fields_to_suppress[self.current_type][field_name] = field_type
+        # At a field of a type that is not the query type
+        # A field must be suppressed if its type depends on an unqueryable type. There are two ways
+        # for a type to become unqueryable: either the type itself was suppressed, or the type is an
+        # interface and another type implementing the interface was suppressed. The field can either
```

Comment:
nit: I find bulleted lists so much easier to read than paragraphs of text, especially when the list items are long (like the second bullet). Your list is short enough that in paragraph is probably fine, but this is a little more readable to me.
```suggestion
        # for a type to become unqueryable: 
        #    - the type itself was suppressed
        #    - the type is an interface and another type implementing the interface was suppressed
        # The field can either
```